### DISCUSSION TITLE
Problems setting up my first cluster, pleas help.

### DISCUSSION THREAD

**Discussion by BondByteBlaster** [datalust/seq-tickets#2105] [Help]
Hi,

I'm setting up a SEQ cluster on two Windows Server 2022 nodes without a load balancer for now, so I plan to point the canonicalUri to the follower node temporarily. This setup will be revised later.

After a fresh SEQ installation on node 1 (without starting the service), I followed the guide and executed these commands:

```
PS C:\Windows\system32> seq show-key --generate
PS C:\Windows\system32> seq config create
PS C:\Windows\system32> seq config set -k storage.secretKey -v "<generated_key>"
PS C:\Windows\system32> seq secret set -k metastore.msSql.connectionString -v "Server=myapp.cloudproservice.se,15003;Database=seq;User Id=sequser;Password=<redacted>;"
```

However, I encountered the following exception when setting the SQL connection string:

```
[12:27:47 FTL] Could not set the configuration field override
System.Security.Cryptography.CryptographicException: Padding is invalid and cannot be removed.
```

I've reinstalled SEQ and regenerated the key with no success. I've verified network connectivity to the MSSQL database (using Test-NetConnection and other methods), and there seem to be no issues there. Could you please help me understand why I'm getting this exception? Is there a known issue or step I might be missing?

Thank you for your assistance.

**Comment by KodrAus** [comment]
Hi @BondByteBlaster :wave: When installing Seq through the MSI it will typically initialize and start Seq as a service unless you explicitly ask it not to. It looks like that might have been the case here and Seq already has config using the originally generated secret key, which prevents it from being able to decrypt values based on the old key.

If you run:

```
seq service stop
```

then delete `C:\ProgramData\Seq` and rerun those commands does it still fail in the same way?

**Comment by BondByteBlaster** [comment]
Thanks! I did uncheck the "configurate and start" SEQ service but it apperas it did some configurations anyway.

Now I manage to run all the commands on node1, it saysa that its starting up but infact its not. When I check the log file I find this : 

> {"@t":"2024-02-27T10:05:54.9428590Z","@mt":"Seq {SeqVersion} running on OS {OSCaption}","SeqVersion":"2024.1.11028","OSCaption":"Microsoft Windows Server 2022 Datacenter","SourceContext":"Seq.Server.Features.Diagnostics.EnvironmentInformationWriter"}
> {"@t":"2024-02-27T10:05:54.9892359Z","@mt":"Running in Windows Service host"}
> {"@t":"2024-02-27T10:05:54.9998203Z","@mt":"Configuring HTTP.sys web server","SourceContext":"Seq.Server.Web.Host.WebHostBuilderExtensions"}
> {"@t":"2024-02-27T10:05:55.2751533Z","@mt":"The Seq server could not be started","@l":"Fatal","@x":"System.UriFormatException: Invalid URI: The format of the URI could not be determined.\r\n   at System.Uri.CreateThis(String uri, Boolean dontEscape, UriKind uriKind, UriCreationOptions& creationOptions)\r\n   at System.Uri..ctor(String uriString)\r\n   at Seq.ServiceProcess.ServerService..ctor(IApplicationContainer applicationContainer, IWebHost host, SeqApiConfig apiConfig, SystemServices systemServices, WorkerPool workerPool, WebSocketHandler webSocketHandler)\r\n   at Seq.Server.Web.Host.AspNetCoreBootstrapper.BuildServer(String instanceName, StorageLayout storageLayout, SeqConfig config, TelemetryClient telemetryClient, IFilesystemCertificateStore certificateStore, IFeatureFlags featureFlags)\r\n   at Seq.Cli.RunCommand.<>c__DisplayClass6_0.<RunAsync>g__Bootstrap|0()\r\n   at Seq.ServiceProcess.SeqWindowsService.<>c__DisplayClass2_0.<.ctor>b__0()"}


Node 2 server is running and are accassable but no SEQ service is configured\installed there yet.

The Seq.json looks like this : 

> {
>   "api": {
>     "canonicalUri": "http://192.168.73.4:5341",
>     "corsAllowedOrigins": [],
>     "frameAncestors": [],
>     "hardSessionExpirySeconds": null,
>     "hstsIncludeSubDomains": false,
>     "hstsMaxAge": 31536000,
>     "idleSessionExpirySeconds": 172800,
>     "ingestionPort": null,
>     "integratedAuthenticationScheme": null,
>     "listenUris": [
>       "http://localhost:5341"
>     ],
>     "redirectHttpToHttps": false,
>     "minRequestBodyDataRateBytesPerSecond": 240.0,
>     "minRequestBodyDataRateGracePeriodMilliseconds": 5000,
>     "webServer": null,
>     "globalRateLimit": null,
>     "globalQueueLimit": null,
>     "ingestRateLimit": null,
>     "ingestQueueLimit": null,
>     "queryRateLimit": null,
>     "queryQueueLimit": null
>   },
>   "appHost": {
>     "installPolicy": null,
>     "executionPolicy": null
>   },
>   "audit": {
>     "auditServerUrl": null,
>     "auditServerApiKey": null,
>     "auditPath": null
>   },
>   "certificates": {
>     "certificatesPath": null,
>     "defaultPassword": null
>   },
>   "diagnostics": {
>     "internalLogPath": "C:\\ProgramData\\Seq\\Logs",
>     "internalLoggingLevel": "Verbose",
>     "internalLogServerApiKey": null,
>     "internalLogServerUrl": null,
>     "metricsSamplingIntervalSeconds": 300,
>     "nativeStorageMetricsSamplingIntervalMinutes": 60,
>     "telemetryServerUrl": null
>   },
>   "features": {
>     "enabled": []
>   },
>   "firstRun": {
>     "adminPasswordHash": null,
>     "adminUsername": null,
>     "requireAuthenticationForHttpIngestion": false
>   },
>   "metastore": {
>     "msSql": {
>       "connectionString": null,
>       "msiResource": null,
>       "msiTenantId": null,
>       "schema": null
>     },
>     "postgres": {
>       "connectionString": null,
>       "schema": null
>     }
>   },
>   "process": {
>     "workingSetLimitBytes": null
>   },
>   "secretKey": {
>     "provider": null,
>     "providerArgs": null
>   },
>   "services": {
>     "servicesBaseUri": null
>   },
>   "storage": {
>     "secretKey": "<redacted>",
>     "flare": {
>       "concurrentReaderLimit": null,
>       "readerParallelismLimit": null,
>       "indexerPriority": 0.0
>     },
>     "disableFreeDiskSpaceChecking": false,
>     "maximumFutureTimestampDriftSeconds": 3420
>   }
> }

I have tested using both hostnames and ip but its still the same exception?

There is no tables created in the SQL database so maybe its the connection string that it does not like? But if I pick the information from it and try to connect with MS SQL Manager, then it works just great? I can see that the user have correct rights to the SEQ database.

Sugestions?

**Reply by liammclennan** [reply]
As you have two Seq instances, could it be that the error is from a different server to the one with that config file?

It seems very likely that the server that is crashing does not have that configuration.

**Comment by BondByteBlaster** [comment]
I have two nodes intended for a SEQ cluster and a third node where the MS SQL database is running. Currently, I am only working on node 1; nothing has been done on node 2 so far, not even a SEQ installation. All nodes are running Windows Server 2022.

At node 1, I've been trying to install SEQ as a cluster leader, which has proven to be more difficult than expected. I have followed the guide on this page: https://docs.datalust.co/docs/setting-up-a-disaster-recovery-dr-instance (starting from "If you're starting with a clean server").

To be clear, there is only one SEQ instance about to be set up on node 1. The second instance (follower) will be installed on node 2 as soon as node 1 is running properly.

This is what I have tried (several times):

1. Created a database and seqDB user at node 3.
2. Downloaded and installed the latest SEQ version on node 1, making sure to uncheck the "configure and start service."
3. Deleted files under C:\ProgramData\Seq.
4. Generated a new storage key using seq show-key --generate.
5. Created a new configuration with seq config create.
6. Set the storage secret key with seq config set -k storage.secretKey -v "<generated key from step 4>."
7. Set the metastore MS SQL connection string with seq secret set -k metastore.msSql.connectionString -v "Server=myapp.cloudproservice.se,15003;Database=seq;User Id=sequser;Password=<retracted>."
8. Set the API canonical URI with seq config set -k api.canonicalUri -v "node2 Hostname."
9. Ran the node setup command as follows, noting that node01 is the hostname of node 1. I also tried replacing IP addresses with hostnames (node01 and node02).

seq node setup `
  --cluster-listen ws://192.168.54.40:5344 `
  --internal-api http://192.168.54.40 `
  --peer-cluster ws://192.168.54.41:5344 `
  --peer-internal-api http://192.168.54.41 `
  -k "<retracted password (HEX key 128 bit)>" `
  --node-name node01

10. Attempted to start the SEQ service with seq service start.

After attempting to start the service, I found that the SEQ service stopped shortly after without any response when trying to access the SEQ web page. The SEQ log reports a System.UriFormatException: Invalid URI: The format of the URI could not be determined.

I've double-checked that the nodes have access to each other, including a port check, but I cannot find any URI that might be incorrectly formatted.

I'm stuck and considering starting this environment without a SEQ cluster due to these challenges.

**Reply by nblumhardt** [reply]
Hi! I think the problem is likely to be:

> 8. Set the API canonical URI with seq config set -k api.canonicalUri -v "node2 Hostname."

This needs to be a full URI including the `https://` etc.

We're happy continuing to help here, but kicking off an email thread with `support@datalust.co` may make it easier to share more specifics. Fingers crossed this gets your node up and running, though!
